### ISSUE TITLE
[1.3.0-RC1] Update hangs when resolving against Artifactory with auth

### ISSUE THREAD

**Issue by avdv** [sbt/sbt#4700]
- [x] I've read the [CONTRIBUTING](https://github.com/sbt/sbt/blob/develop/CONTRIBUTING.md) guidelines :heavy_check_mark: 

## steps

1. start `sbt`

## problem

It starts to resolve dependencies, but never finishes:
```
[info] Loading project definition from /home/claudio/src/check/project
[info] Updating 

  | => check-build / update 516s
```

## expectation

It should successfully resolve dependencies.

Adding `useCoursier := false` to `project/build.sbt` resolves this problem.

Is it possible to debug into this?

We are using [dynamic revisions](https://ant.apache.org/ivy/history/2.3.0/ivyfile/dependency.html), ie. `2.7.+`, for our own SBT plugin in `project/plugins.sbt` which might coursier to fail...

## notes

sbt version: 1.3.0-RC1

**Comment by eed3si9n** [comment]
@avdv Thanks for the report. Do you have a project that we can use to reproduce this issue on our machine?

**Comment by avdv** [comment]
I could try to reduce it to some minimal reproducer... 

But, I just discovered that it is indeed trying to resolve the dynamic revision against our internal Artifactory server, **verbatim**:
```
https://artifactory.internal.com/ivy-virtual/com.sample/eve-lib_2.12/2.7.+/ivys/ivy.xml
```

**Comment by avdv** [comment]
BTW, does the coursier resolver work with credentials out of the box? Some months ago, one had to explicitly set `coursierUseSbtCredentials := true` or something like that...

**Comment by eed3si9n** [comment]
> BTW, does the coursier resolver work with credentials out of the box? 

It should. I am translating the credentials here:

https://github.com/sbt/sbt/blob/v1.3.0-RC1/main/src/main/scala/sbt/internal/librarymanagement/CoursierInputsTasks.scala#L208

**Comment by ikryvorotenko** [comment]
I've got the same issue on my repo. And It has the plugin dependency coming from internal artifactory, just as @avdv described.

It looks like I have faced this issue before when tried to introduce Coursier plugin, but that time I just decided to remove it.

@avdv @eed3si9n  how do you trace/debug sbt to see more what's going on inside?

**Comment by eed3si9n** [comment]
You can try `sbt -debug`

**Comment by phdoerfler** [comment]
Is this in any way related to https://github.com/coursier/coursier/issues/1176?

**Comment by avdv** [comment]
> Is this in any way related to [coursier/coursier#1176](https://github.com/coursier/coursier/issues/1176)?

Probably not.

This issue here seems to be a problem with coursier starting with 1.1.0-M14, it hangs when resolving dynamic revisions (with credentials and/or against Artifactory -- I still need to pinpoint the problem).

Using the versions before (1.1.0-M8 <> 1.1.0-M13-4) gave a resolution error, which is not the expected output either but OK.

**Comment by ikryvorotenko** [comment]
@avdv It seems like the root cause of the issue is custom artifactory server authentication. In my project, I have a strict dependency version of an sbt-plugin coming from private artifactory and it still hangs.

**Comment by avdv** [comment]
> It seems like the root cause of the issue is custom artifactory server authentication. In my project, I have a strict dependency version of an sbt-plugin coming from private artifactory and it still hangs.

Yes, you're right. I forgot about the global resolver configuration. It only hangs when trying to resolve against Artifactory with credentials. I'll update the coursier issue.

**Comment by xerial** [comment]
I hit the same problem when using a private Artifactory repository that requires password. 
Before sbt 1.3.0-RC1, sbt-coursier with `coursierUseSbtCredentials := true` configuration worked for me.

**Comment by alexarchambault** [comment]
https://github.com/coursier/coursier/issues/1179 seems fixed, so the issue here should be fixed with sbt `1.3.0-RC2`.

**Comment by eed3si9n** [comment]
Thanks for the update.

**Comment by avdv** [comment]
up-update: the upstream issue is apparently not yet fixed

**Comment by ikryvorotenko** [comment]
yea, this is still reproducible on my project too

**Comment by joan38** [comment]
Hitting it in SBT 1.3.0-RC3. Can we reopen the issue?
Here are the reproduction steps:
```
echo "useCoursier := false" > build.sbt
mkdir project
echo "sbt.version=1.3.0-RC3" > project/build.properties
echo "addSbtPlugin(\"ch.epfl.scala\" % \"sbt-scalafix\" % \"0.9.5\")" > project/plugins.sbt
sbt updateSbtClassifiers
```

**Comment by eed3si9n** [comment]
@joan38 How is your repro steps related to resolution with authentication?

**Comment by joan38** [comment]
@eed3si9n sorry that is indeed unrelated. I landed here via some Googling.
My issue is if we do `useCoursier := false` in build.sbt we need to do it in project/plugins.sbt
It's minor, should I open another issue?

**Comment by eed3si9n** [comment]
@joan38 Yea. Could you open a new issue? I'll close this one.

**Comment by avdv** [comment]
> yea, this is still reproducible on my project too

FYI, this issue is indeed resolved for me with 1.3.0-RC3. :tada: :sweat_smile:
